### ISSUE TITLE
Matching provider's headers

### ISSUE THREAD

**Issue by metaflow** [DiUS/pact-jvm#437]
I am trying to create a pact when provider is free to specify any `content-type` but have to provide one.

```
.consumer("wget")
.hasPactWith("server")
.given("file exist on server")
.uponReceiving("get file")
.method("GET")
.matchPath("/.*", "/{file-path}")
.willRespondWith()
.status(200)
.matchHeader("Content-Type", ".*", "application/pdf")  // <-----------
```
part of generated json:
```
...
"response": {
  "status": 200,
  "headers": {
     "Content-Type": "application/pdf"
  },
  "matchingRules": {
    "$.headers.Content-Type": {
      "regex": ".*"
    }}},
```
I am running pact-jvm-provider-junit test with http target.

Test passes but comment on match is 
```
      includes headers
        "Content-Type" with value "application/pdf" (OK)
```
while it is not true and it's better to indicate that it matched `.*`. But that is not a real issue.

----

Now, if I am adding body matcher like ` .body(PactDslJsonRootValue.stringMatcher(".*", "any"))` then several things broke:
1. in json it states `"body": "\"any\"",` (it's probably similar to #385)
2. header stops matching and fails with `Expected a response type of 'application/xml' but the actual type was 'application/pdf'` (server responded with xml all that time).

**Comment by uglyog** [comment]
The `Content-Type` header is a bit special in that it is used to determine what matching implementation to apply. So changing it has the side effect that you change the matching. The DSLs (like `.body(PactDslJsonRootValue.stringMatcher(".*", "any"))`) work with specific content types and change the expected header, so the matcher you setup is probably being removed.

**Comment by uglyog** [comment]
I also agree that the comment on match should reflect that a matcher was used, instead of the example value.
